### PR TITLE
Feature/stricter auth checks

### PR DESCRIPTION
**Pull request by larryk85**
Added stricter checks for action/context-free action and transaction authorities.  
#1766

### CODE REVIEW

File: libraries/chain/include/eosio/chain/transaction_metadata.hpp [hunk 1]
```diff
@@ -77,3 +79,4 @@ class transaction_metadata {
 
 } } // eosio::chain
 
+FC_REFLECT( eosio::chain::transaction_metadata, (raw_trx)(signing_keys)(id)(region_id)(cycle_index)(shard_index)(bandwidth_usage)(published)(sender)(sender_id))
```
Comment (brianjohnson5972) [inline libraries/chain/include/eosio/chain/transaction_metadata.hpp]: Should is_implicit be added to reflection?

Comment (larryk85) [reply libraries/chain/include/eosio/chain/transaction_metadata.hpp]: Yes, it should.

**Comment by gleehokie** [comment]
Dan is not sure we want these checks, too restrictive.

**Comment by gleehokie** [comment]
Dan and Bucky talked, we're going ahead and merging.

**Comment by gleehokie** [comment]
For some reason Jenkins tried to mark the test as passed but GitHub rejected it.  Appears to be a GH issue.  Going ahead and merging.

**Comment by andriantolie** [comment]
The related issue is #1766 and it passes

**Pull request merged**